### ISSUE TITLE
Self Heating

### ISSUE THREAD

**Issue by dbrgn** [coredump-ch/photon-sensor-board#1]
The sensor can heat up if it is under constant load.

Instead, connect it to the Photon through something like a transistor so it can be turned on and off.

**Comment by rnestler** [comment]
Isn't it a digital sensor? I should support some kind of sleep mode then were it doesn't heat up.

**Comment by dbrgn** [comment]
Yes, it's one-wire. A measurement only takes place when we ask for it, but apparently just having power applied could still heat it a little.

I'm not completely sure though, I just want to rule out all possibilities of self heating.

The worst self-heating factor was the distance from the board. When connecting a sensor directly to the PCB, the readings are 4-5°C higher than they should be. On contrast if you put a 10cm wire between the sensor and the board, they're mostly accurate.

Another factor is that we're polling quite frequently. For the water temperature sensors that won't be a problem, as we'll only poll them hourly or so. But when polling every 2-3 seconds, it can probably heat up from the load.

Here are some interesting links:

http://arduino.stackexchange.com/questions/789/my-ds18b20-is-reading-high-how-can-i-get-it-to-return-the-correct-temperature

http://forum.arduino.cc/index.php?topic=214113.0

**Comment by dbrgn** [comment]
This is also interesting, wasn't aware of that:

> You need one inside a metal package if you want to measure air temperature, otherwise they pick up radiant heat from nearby objects (people) and read upto several degrees out - air temperature measurement is confounded by the radiant  heating.
> 
> Black bodies absorb heat radiation efficiently, metals don't.

(From the Arduino thread)

**Comment by rnestler** [comment]
Yes measuring temperature isn't easy....

Regarding the self heating: We could also mathematically compensate it if we can determine it precisely enough.

**Comment by dbrgn** [comment]
I think the main thing we should do is testing.

Do you have calibrated, reliable, standalone temperature sensors at Sensi that could be used for comparison and testing? Then we could test things like distance from the PCB, frequency of polling, etc.
